### PR TITLE
AB#87742 Site - Landing Page and Name views - Handlers

### PR DESCRIPTION
**Pull request by Carq**

### CODE REVIEW

File: src/HE.Investment.AHP.Contract/HE.Investment.AHP.Contract.csproj [hunk 1]
```diff
@@ -20,6 +20,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Dawn.Guard" Version="1.12.0" />
```
Comment (Antek012) [inline src/HE.Investment.AHP.Contract/HE.Investment.AHP.Contract.csproj:23]: This package is deprecated and no longer maintained.

File: src/HE.Investment.AHP.Contract/Site/ValueObjects/SiteId.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+using Dawn;
+using HE.Investments.Common.Domain;
+using HE.Investments.Common.Extensions;
+
+namespace HE.Investment.AHP.Contract.Site.ValueObjects;
+
+public class SiteId : ValueObject
```
Comment (Antek012) [inline src/HE.Investment.AHP.Contract/Site/ValueObjects/SiteId.cs:7]: please inherit from StringIdValueObject

File: src/HE.Investment.AHP.Domain/Site/CommandHandlers/SiteBaseCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    protected async Task<OperationResult> Perform(Func<SiteEntity, Task> action, SiteId siteId, CancellationToken cancellationToken)
+    {
+        var userAccount = await AccountUserContext.GetSelectedAccount();
+        var site = await SiteRepository.GetSite(siteId, userAccount, cancellationToken);
+
+        try
+        {
+            await action(site);
+        }
+        catch (DomainValidationException domainValidationException)
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/Site/CommandHandlers/SiteBaseCommandHandler.cs:35]: This should not be needed as we have DomainValidationHandler.

File: src/HE.Investment.AHP.WWW.Tests/HE.Investment.AHP.WWW.Tests.csproj [hunk 4]
```diff
@@ -51,6 +51,7 @@
   <ItemGroup>
     <ProjectReference Include="..\HE.Investment.AHP.WWW\HE.Investment.AHP.WWW.csproj" />
     <ProjectReference Include="..\HE.Investments.Common.WWWTestsFramework\HE.Investments.Common.WWWTestsFramework.csproj" />
+    <ProjectReference Include="..\HE.Investments.IntegrationTestsFramework\HE.Investments.IntegrationTestsFramework.csproj" />
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW.Tests/HE.Investment.AHP.WWW.Tests.csproj:54]: Why do we need Integration tests to unit tests coupling?

**Review by Antek012 — APPROVED** [review]

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.Contract/HE.Investment.AHP.Contract.csproj (hunk 1 above)
Comment (Carq) [reply src/HE.Investment.AHP.Contract/HE.Investment.AHP.Contract.csproj:23]: True, but it is very simple and does exactly what we need.

File: src/HE.Investment.AHP.WWW.Tests/HE.Investment.AHP.WWW.Tests.csproj (hunk 4 above)
Comment (Carq) [reply src/HE.Investment.AHP.WWW.Tests/HE.Investment.AHP.WWW.Tests.csproj:54]: A lot of helpers for HtmlDocument are defined there. I will extract them to separate project. I will this in the next CR.

File: src/HE.Investment.AHP.Domain/Site/CommandHandlers/SiteBaseCommandHandler.cs (hunk 3 above)
Comment (Carq) [reply src/HE.Investment.AHP.Domain/Site/CommandHandlers/SiteBaseCommandHandler.cs:35]: Nice. Fixed.

**Pull request merged**